### PR TITLE
Fix issues with Doxygen 1.8.17

### PR DESCRIPTION
**Pull request by utzig**
Fix warnings that show up when running 1.8.17, update config, fix mismatched groups, duplicated or invalid usage of params/retvals, remove \copydetails usage.

/cc @thst-nordic - This should work OK on both 1.8.13 and 1.8.17

### CODE REVIEW

File: include/bluetooth/mesh/gen_prop_cli.h [hunk 1]
```diff
@@ -181,6 +185,10 @@ int bt_mesh_prop_cli_prop_get(struct bt_mesh_prop_cli *cli,
  * not configured.
  * @retval -EAGAIN The device has not been provisioned.
  * @retval -ETIMEDOUT The request timed out without a response.
+ *
+ * The User Property may only be set if the server enabled user write access to
+ * it. If this is not the case, the server will only respond with the set user
+ * access mode for the given property.
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/gen_prop_cli.h:191]: Looks like this was pasted one time too many

File: include/bluetooth/mesh/sensor_cli.h [hunk 2]
```diff
@@ -602,7 +602,6 @@ int bt_mesh_sensor_cli_series_entry_get(
  *                         has been changed to the number of columns in the
  *                         response.
  *  @retval -ENOTSUP       The sensor doesn't support series data.
- *  @retval -ENODEV        The sensor server doesn't have the given sensor.
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/sensor_cli.h:605]: Suggest changing this to `-ENOENT` instead of removing it

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by grochu — APPROVED** [review]
Looks good.

File: include/bluetooth/mesh/sensor_cli.h (hunk 2 above)
Comment (utzig) [reply include/bluetooth/mesh/sensor_cli.h:605]: Oh, I did not just remove it for no reason, the **exact** same line with the **exact** same content also exists a few lines above.

File: include/bluetooth/mesh/gen_prop_cli.h (hunk 1 above)
Comment (utzig) [reply include/bluetooth/mesh/gen_prop_cli.h]: Oops, fixed!

File: include/bluetooth/mesh/sensor_cli.h (hunk 2 above)
Comment (trond-snekvik) [reply include/bluetooth/mesh/sensor_cli.h:605]: Oh, I see, my bad

**Review by trond-snekvik — APPROVED** [review]

**Review by hurricaneFromMoscow — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by thst-nordic** [comment]
do we need to remove the 'DNM' commit before merging?

**Comment by utzig** [comment]
> do we need to remove the 'DNM' commit before merging?

This PR must be updated after https://github.com/nrfconnect/sdk-nrfxlib/pull/333 is merged, then the label can be removed.

**Pull request merged**